ts-lib/tests/contract.test.ts

```diff
@@ -65,4 +65,17 @@ describe("EthKeys contract tests", () => {
     );
     expect(decryptedKey).toBe(ethWallet.privateKey);
   });
+
+  it("CryptoJS-AES RoundTrip", async () => {
+    const keyManager = new CryptoJSKeyManager(contract);
+    const ethWallet = ethers.Wallet.createRandom();
+
+    await keyManager.encryptAndSetKey(ethWallet, privateKey);
+
+    const decryptedKey = await keyManager.retrieveAndDecryptKey({
```

Comment:
Let's make this API "symmetrical", so if we pass `EthWallet` to `encryptAndSet`, we should get back `EthWallet` from `retrieveAndDecrypt`, same for the XSalsa encryption

Reply:
This is a good idea. Lets see if there is a way to construct an EthWallet object from just a private key.

Reply:
I will have to do a bit here to determine exactly what should be passed in and out of this function.

calling ethers.Wallet.createRandom() comes with a lot more info that just `new ethers.Wallet(privateKey)`

Example

```
    Before HDNodeWallet {
      provider: null,
      address: '0xF3ddfF45A25C9464c0489dD88dbF74BbF18Bf175',
      publicKey: '0x02a7f5e094f538b2676cc306e536035880bd24d785073769a4a9c862e31700d6da',
      fingerprint: '0xe6c847e5',
      parentFingerprint: '0x60194bc1',
      mnemonic: Mnemonic {
        phrase: 'leader film divide garage trust venture produce when remind become clog skate',
        password: '',
        wordlist: LangEn { locale: 'en' },
        entropy: '0x7e8ac8ffafbe9be4aaefd1b5c27cae65'
      },
      chainCode: '0xfe609a47f6a94ebbdedc28df21d285e883616fef683d0f841478e788f4880a4b',
      path: "m/44'/60'/0'/0/0",
      index: 0,
      depth: 5
    }
```

```
    After Wallet {
      provider: null,
      address: '0xF3ddfF45A25C9464c0489dD88dbF74BbF18Bf175'
    }
```

Reply:
What might be better is not passing ethers.Wallet object, but merely the privateKey. Then we can easily assert the length and format as well. Currently the decryption method returns only the privateKey anyway.

Reply:
I think this suggestion would be more suitable to change after this PR is merged (since everything is working as expected and its going to be a bigger change).

Reply:
Please see #10 for changes related to this.

Reply:
Since the linked PR was approved, I will assume this is ok too.